apps/common/behaviours/information-you-have-given-us.js

```diff
@@ -0,0 +1,60 @@
+const { findSatisfiedForkCondition } = require('../../../utils');
+
+const buildStepJourneyFromSessionValues = req => {
+  const formSteps = Object.keys(req.form.options.steps);
+  const stepJourney = [formSteps[0]];
+
+  const addNextStep = steps => {
+    const currentStep = steps[steps.length - 1];
+    const forksInStep = req.form.options.steps[currentStep].forks;
+    const satisfiedForkCondition = findSatisfiedForkCondition(req, forksInStep);
+    const nextStep = satisfiedForkCondition ? satisfiedForkCondition.target : req.form.options.steps[currentStep].next;
+    if (nextStep) {
```

Comment:
I would suggest to add a check to avoid infinite recursion
```suggestion
    if (nextStep && !stepJourney.includes(nextStep)) {
```